### ISSUE TITLE
Fruit tree crash during worldgen/exploration on 1.17.0-pre.5

### ISSUE THREAD

**Issue by syntaxaire** [anegostudios/VintageStory-Issues#2041]
**Game Version:** 1.17.0-pre.5
**Platform:** Windows
**Modded:** Yes (HUD Clock, ProspectorInfo)
**SP/MP:** Singleplayer

### Description
Crash during worldgen or ticking.

### How to reproduce
This crash was encountered while flying in creative to explore a map and generating terrain. The seed was `-937092196` with default worldgen settings except climate set to 'warm', and I was generating terrain around 3000,1,1000 with max view distance.

**Note: time is frozen in this game, nothing should be ticking.**
`/time speed 0` was used before doing any worldgen or exporation.

### Logs
```
Running on 64 bit Windows with 32 GB RAM 
Game Version: v1.17.0-pre.5 (Unstable)
Loaded Mods: game@1.17.0-pre.5, hudclock@2.5.0, prospectorinfo@3.1.2, creative@1.17.0-pre.5, survival@1.17.0-pre.5
2022-07-23 11:34:18 AM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityFruitTreeBranch.updateProperties() in C:\Users\Tyron\Documents\vintagestory\game\VSSurvivalMod\Systems\FruitTree\BEFruitTreeBranch.cs:line 213
   at Vintagestory.GameContent.BlockEntityFruitTreeBranch.FromTreeAttributes(ITreeAttribute tree, IWorldAccessor worldForResolving) in C:\Users\Tyron\Documents\vintagestory\game\VSSurvivalMod\Systems\FruitTree\BEFruitTreeBranch.cs:line 365
   at Vintagestory.Client.NoObf.ClientChunk.AddOrUpdateBlockEntityFromPacket(Packet_BlockEntity p, ClientMain game) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Model\ClientChunk.cs:line 409
   at Vintagestory.Client.NoObf.GeneralPacketHandler.HandleBlockEntities(Packet_Server packet) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Systems\GeneralPacketHandler.cs:line 808
   at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ClientMain.cs:line 1138
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 123
   at Vintagestory.Client.ScreenManager.Render(Single dt) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ScreenManager.cs:line 664
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ScreenManager.cs:line 608
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 125
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 476
   at OpenTK.GameWindow.DispatchRenderFrame() in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 452
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 375
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs)
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 87
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 2022-07-02 5:33:47 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1169322480264892805, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.16.5.0
P3: 10.0.19043.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\User\AppData\Local\Temp\RDR5AF7.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5B07.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5B18.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5B25.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5B46.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: 9c49573f-e84c-489a-afa2-374acf1b0273
Report Status: 268435456
Hashed bucket: 66b2146b18168122903a4499b8115185
Cab Guid: 0 }
--------------
{ TimeGenerated = 2022-01-08 2:31:39 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 2137399946764805085, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.16.0.0
P3: 10.0.19043.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\User\AppData\Local\Temp\RDR4F5.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER506.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER507.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER50F.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER53F.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: b4c1fd7b-f16f-4c84-a442-665bd0d6e8a0
Report Status: 268435456
Hashed bucket: 90206cb8882a62a92da991ed59dae3dd
Cab Guid: 0 }
--------------
{ TimeGenerated = 2021-12-04 2:11:10 PM, Site = , Source = Application Hang, Message = The program Vintagestory.exe version 1.16.0.0 stopped interacting with Windows and was closed. To see if more information about the problem is available, check the problem history in the Security and Maintenance control panel.

Process ID: 706c

Start Time: 01d7e92cc153f8be

Termination Time: 14

Application Path: C:\Users\User\AppData\Roaming\Vintagestory\Vintagestory.exe

Report Id: 3585f597-2ca8-462c-8780-fbbdeb917592

Faulting package full name: 

Faulting package-relative application ID: 

Hang type: Unknown
 }
```

**Comment by syntaxaire** [comment]
I was able to duplicate this crash by flying in a different direction and just hanging out for a while with view distance turned up.

**Comment by tyronx** [comment]
strange. thanks for the report. next version should at least prevent the crash.

**Comment by syntaxaire** [comment]
I regenerated this world with the same seed, no mods, and I'm seeing a second type of fruit tree crash near spawn quite quickly.

```
Running on 64 bit Windows with 32 GB RAM 
Game Version: v1.17.0-pre.5 (Unstable)
Loaded Mods: game@1.17.0-pre.5, creative@1.17.0-pre.5, survival@1.17.0-pre.5
2022-07-24 2:29:06 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityFruitTreeBranch.GenMeshes() in C:\Users\Tyron\Documents\vintagestory\game\VSSurvivalMod\Systems\FruitTree\BEFruitTreeBranch.cs:line 292
   at Vintagestory.GameContent.BlockEntityFruitTreeBranch.GenMesh() in C:\Users\Tyron\Documents\vintagestory\game\VSSurvivalMod\Systems\FruitTree\BEFruitTreeBranch.cs:line 221
   at Vintagestory.GameContent.BlockEntityFruitTreeBranch.FromTreeAttributes(ITreeAttribute tree, IWorldAccessor worldForResolving) in C:\Users\Tyron\Documents\vintagestory\game\VSSurvivalMod\Systems\FruitTree\BEFruitTreeBranch.cs:line 365
   at Vintagestory.Client.NoObf.ClientChunk.AddOrUpdateBlockEntityFromPacket(Packet_BlockEntity p, ClientMain game) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Model\ClientChunk.cs:line 409
   at Vintagestory.Client.NoObf.GeneralPacketHandler.HandleBlockEntities(Packet_Server packet) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Systems\GeneralPacketHandler.cs:line 808
   at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ClientMain.cs:line 1138
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 123
   at Vintagestory.Client.ScreenManager.Render(Single dt) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ScreenManager.cs:line 664
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ScreenManager.cs:line 608
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 125
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 476
   at OpenTK.GameWindow.DispatchRenderFrame() in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 452
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 375
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs)
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 87
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 2022-07-02 5:33:47 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1169322480264892805, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.16.5.0
P3: 10.0.19043.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\User\AppData\Local\Temp\RDR5AF7.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5B07.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5B18.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5B25.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5B46.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: 9c49573f-e84c-489a-afa2-374acf1b0273
Report Status: 268435456
Hashed bucket: 66b2146b18168122903a4499b8115185
Cab Guid: 0 }
--------------
{ TimeGenerated = 2022-01-08 2:31:39 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 2137399946764805085, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.16.0.0
P3: 10.0.19043.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\User\AppData\Local\Temp\RDR4F5.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER506.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER507.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER50F.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER53F.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: b4c1fd7b-f16f-4c84-a442-665bd0d6e8a0
Report Status: 268435456
Hashed bucket: 90206cb8882a62a92da991ed59dae3dd
Cab Guid: 0 }
--------------
{ TimeGenerated = 2021-12-04 2:11:10 PM, Site = , Source = Application Hang, Message = The program Vintagestory.exe version 1.16.0.0 stopped interacting with Windows and was closed. To see if more information about the problem is available, check the problem history in the Security and Maintenance control panel.

Process ID: 706c

Start Time: 01d7e92cc153f8be

Termination Time: 14

Application Path: C:\Users\User\AppData\Roaming\Vintagestory\Vintagestory.exe

Report Id: 3585f597-2ca8-462c-8780-fbbdeb917592

Faulting package full name: 

Faulting package-relative application ID: 

Hang type: Unknown
 }
```

**Comment by syntaxaire** [comment]
@tyronx Another case of the second crash in `BlockEntityFruitTreeBranch.GenMeshes()` was posted on Discord with rc2: <https://discord.com/channels/302152934249070593/334233095374897152/1001582838543560714>

**Comment by noelle-lavenza** [comment]
Assuming this was from `capi.Tesselator.TesselateShape("fruittreebranch", nowTesselatingShape, out mesh, this, new Vec3f(cshape.rotateX, cshape.rotateY, cshape.rotateZ), 0, 0, 0, null, selectiveElements?.ToArray());`, if `capi` was null this should be fixed in 1.17.8. Indeed, the change logs mention `Fixed: Should fix a client crash related to fruit trees`.

Duplicate of #2182. potential duplicate of #1953 as well. Both are closed already and I can't reproduce this issue on latest.
